src/v/security/audit/audit_log_manager.cc

```diff
@@ -550,11 +564,25 @@ ss::future<> audit_log_manager::start() {
     });
     if (_audit_enabled()) {
         vlog(adtlog.info, "Starting audit_log_manager");
+        if (!this->enqueue_app_lifecycle_event(
+              application_lifecycle::activity_id::start,
+              ss::sstring{subsystem_name})) {
+            vlog(adtlog.error, "Failed to enqueue audit lifecycle start event");
+            // TODO I should error here, yeah?
```

Comment:
Or just return, right? Or is this supposed to short-circuit startup?

Reply:
yeah that's the question.  Should I throw out of this `start()` method.  If this fails to enqueue, something is very wrong.

Reply:
>  Should I throw out of this start() method. If this fails to enqueue, something is very wrong.

Should failing to enqueue an audit event make the broker to stop?

Reply:
If the audit message fails to be enqueued, that's the result of the sharded buffers already being full.  It would be indicative of a serious issue where auditing would not function.

Reply:
I think we should bring in the others opinion on failing to enqueue application lifecycle events, @deniscoady @dotnwat

Reply:
i haven't thought through this carefully, but my immediate reaction is that failing on startup/shutdown are fairly sensitive times in the application lifecycle.

are we only talking about a few possible events like start / stop? if so, and if enqueue failure mostly means that there isn't enough memory available or something, then perhaps it makes sense to arrange for the app life cycle events to never fail for reasons like over quota assuming that we know there will never be more than a couple of them.

Reply:
Lifecycle start events happen in three cases:
1. Redpanda application start
2. Initial start of audit log subsystem (as part of startup)
3. Whenever the audit log subsystem is re-enabled after it's disabled (while running)

The likelihood of 1 or 2 failing is probably slim to none with the only exception I can think of if the audit subsystem is misconfigured.  To prevent that, we should set a sensible minimum limit to the queues to ensure they won't ever be _that_ small.

#3 has a higher possibility (in case the queues are already full and we disable and then re-enable the system).

I'm fine with just logging the an error if this happens at any time, but I also feel that if there is a problem even starting the audit system, then it should be a hard failure.

I only print an error if the application stop events fail to be enqueued.

Reply:
>but I also feel that if there is a problem even starting the audit system

I think it all hinges on what is included in "problem even starting". That seems to be different than the original statement of "failing to enqueue application lifecycle events". I think neither of these are precise right now?

Reply:
If enqueuing a lifecycle start event fails, that would indicate that the sharded buffers are either misconfigured or already full (or maybe some other situation where memory allocation is failing, I don't know).

It would indicate that the ability to enqueue future audit events would fail.  Depending on which audit events are enabled, this could cause cluster to be unusable.

I'm looking for guidance on what to do in this situation.

Reply:
I'm fine with this, we can revisit this if we see issues with it when testing on the LRC, i don't think its a blocker here